server/notebooks/tasks.py

```diff
@@ -0,0 +1,50 @@
+from collections import defaultdict
+from datetime import datetime, timedelta
+
+import pytz
+from spinach import Tasks
+
+from ..settings import NOTEBOOK_REVISION_FIXED_WINDOW_LENGTH_SECS
+from .models import NotebookRevision
+
+tasks = Tasks()
+
+
+@tasks.task(name="notebooks:execute_notebook_revisions_cleanup")
+def execute_notebook_revisions_cleanup(notebook_id, now_utc=None):
+    """Prue revision hisotry.
+
+    Prune revision history so that there is only one revision in each
+    time window, also marking draft revisions that older than window length
+    as non-draft.
+
+    * Time window: This task groups draft revisions into fixed-size windows
+      (alsocalled Tumbling windows).
```

Comment:
I didn't know about tumbling windows. Good catch 😄 I'd probably incorporate this text into the paragraph above, as suggested